application/src/main/java/org/togetherjava/tjbot/commands/github/GitHubCommand.java

```diff
@@ -0,0 +1,123 @@
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.utils.StringDistances;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.*;
+import java.util.function.ToIntFunction;
+import java.util.regex.Matcher;
+import java.util.stream.Stream;
+
+/**
+ * Slash command (/github) used to search for an issue in one of the repositories listed in the
+ * config. It even comes with auto completion!
+ */
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class GitHubCommand extends SlashCommandAdapter {
+    private static final Duration CACHE_EXPIRES_AFTER = Duration.ofMinutes(1);
+
+    /**
+     * Compares the getUpdatedAt values.
+     */
+    private static final Comparator<GHIssue> GITHUB_ISSUE_TIME_COMPARATOR = (i1, i2) -> {
+        try {
+            return i2.getUpdatedAt().compareTo(i1.getUpdatedAt());
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex);
+        }
+    };
+
+    private static final String TITLE_OPTION = "title";
+
+    private final GitHubReference reference;
+
+    private Instant lastCacheUpdate;
+    private List<String> autocompleteGHIssueCache;
+
+    public GitHubCommand(GitHubReference reference) {
+        super("github-search", "Search configured GitHub repositories for an issue/pull request",
+                CommandVisibility.GUILD);
+
+        this.reference = reference;
+
+        getData().addOption(OptionType.STRING, TITLE_OPTION,
+                "Title of the issue you're looking for", true, true);
+
+        updateCache();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        String title = event.getOption(TITLE_OPTION).getAsString();
+        Matcher matcher = GitHubReference.ISSUE_REFERENCE_PATTERN.matcher(title);
+
+        if (!matcher.find()) {
+            event.reply(
+                    "Could not parse your query. Was not able to find an issue number in it (e.g. #207).")
+                .setEphemeral(true)
+                .queue();
+
+            return;
+        }
+
+        reference.findIssue(Integer.parseInt(matcher.group(GitHubReference.ID_GROUP)))
+            .ifPresentOrElse(issue -> event.replyEmbeds(reference.generateReply(issue)).queue(),
+                    () -> event.reply("Could not find the issue you are looking for.")
+                        .setEphemeral(true)
+                        .queue());
+    }
+
+    @Override
+    public void onAutoComplete(CommandAutoCompleteInteractionEvent event) {
+        String title = event.getOption(TITLE_OPTION).getAsString();
+
+        if (title.isEmpty()) {
+            event.replyChoiceStrings(autocompleteGHIssueCache.stream().limit(25).toList()).queue();
```

Comment:
don't use the magic number `25` use [OptionData#MAX_CHOICES](https://javadoc.io/doc/net.dv8tion/JDA/latest/net/dv8tion/jda/api/interactions/commands/build/OptionData.html#MAX_CHOICES) instead